### PR TITLE
Add shadow

### PR DESCRIPTION
**Pull request by mohsen-schibsted**
https://www.figma.com/file/oHBCzDdJxHQ6fmFLYWUltf/Warp---Components-2.0?type=design&node-id=6165-5119&mode=design&t=YYKVaK0zAD6nR29q-0

### CODE REVIEW

File: Sources/Button/WarpButtonStyle/UIMetricsFactory.swift [hunk 1]
```diff
@@ -56,13 +41,10 @@ extension Warp.Button {
...
         var cornerRadius: CGFloat {
-            switch type {
-                case .utility:
-                    return 4
-
-                default:
-                    return 8
+            if type.isUtilityRelatedButton {
+                return 4
             }
```
Comment (vishkaischibsted) [inline Sources/Button/WarpButtonStyle/UIMetricsFactory.swift:46]: Please add an empty line after "}".

File: Sources/Button/WarpButtonStyle/UIMetricsFactory.swift [hunk 2]
```diff
@@ -18,29 +18,14 @@ extension Warp.Button {
...
-            switch type {
-                case .utilityOverlay:
-                    return isEnabled ? 1: 0
-
-                default:
-                    return 0
+        var shadow: Shadow? {
+            if type == .utilityOverlay, isEnabled {
+                return .small
             }
```
Comment (vishkaischibsted) [inline Sources/Button/WarpButtonStyle/UIMetricsFactory.swift:27]: Please add an empty line after "}".

File: Sources/Button/WarpButtonStyle/UIMetricsFactory.swift [hunk 3]
```diff
@@ -18,29 +18,14 @@ extension Warp.Button {
             self.size = size
             self.isEnabled = isEnabled
         }
-
-        /// Button shadow radius.
+        
+        /// Button shadow
```
Comment (vishkaischibsted) [inline Sources/Button/WarpButtonStyle/UIMetricsFactory.swift:22]: Add "." at the end.

File: Sources/Shadow/Shadow.swift [hunk 4]
```diff
@@ -0,0 +1,106 @@
+import SwiftUI
+
+public enum Shadow: String, CaseIterable {
```
Comment (vishkaischibsted) [inline Sources/Shadow/Shadow.swift:3]: Remove `String` conformance, and based on your design remove `CaseIterable` if possible.

File: Sources/Shadow/Shadow.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    case small, medium, large, xLarge
+    
+    var opacity1: Double {
```
Comment (vishkaischibsted) [inline Sources/Shadow/Shadow.swift:6]: Mark all the computed properties as private please.

File: Sources/Shadow/Shadow.swift [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            return 3
+        case .large:
+            return 6
+        case .xLarge:
+            return 9
+        }
+    }
+}
+
+struct ShadowType: ViewModifier {
```
Comment (vishkaischibsted) [inline Sources/Shadow/Shadow.swift:86]: If this should be expose to user,
Mark it as public,
But if this should not be exposed, then mark it as private in order to hide detail implementation from outside of this file.

Also rename it to something more meaningful,
Proposed: ShadowViewModifier

File: Sources/Shadow/Shadow.swift [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    var shadow: Shadow
```
Comment (vishkaischibsted) [inline Sources/Shadow/Shadow.swift:87]: Make it a constant please.

File: Sources/Shadow/Shadow.swift (hunk 4 above)
Comment (mohsen-schibsted) [reply Sources/Shadow/Shadow.swift:3]: I need the `String` for the title of the `picker` on the `demoApp`. Should I make a variable in the `DemoApp` then?

Comment (vishkaischibsted) [reply Sources/Shadow/Shadow.swift:3]: @mohsen-schibsted Exactly, lets encapsulate the logics related to `DemoApp` inside of it.
The same as `CaseIterable`, who needs to add all the shadows to a single view in consumer side.

File: SampleApp/ShadowView.swift [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+extension Warp.Shadow {
+    fileprivate static var allCases: [Warp.Shadow] = [
+        .small,
+        .medium,
+        .large,
+        .xLarge
+    ]
+    
+    fileprivate var rawValue: String {
```
Comment (vishkaischibsted) [inline SampleApp/ShadowView.swift:53]: `rawValue` is a bit vague,
`name` or `title` would be more appropriate.

Feel free to ignore this comment.

**Review by vishkaischibsted — APPROVED** [review]

**Pull request merged**